src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +114,60 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    // For each amendment this version supports, construct the AmendmentState
+    // object by calling addKnown. Set any vetoes or defaults. A pointer to the
+    // AmendmentState can be stashed
+
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        toAdd.emplace (uint256 (_.first), _.second);
+    }
+
+    {
+        // add the amendments from the config file
+        auto const& section = getConfig ().section (SECTION_AMENDMENTS);
+        int const numExpectedToks = 2;
+        for (auto const& _ : section.lines ())
+        {
+            boost::tokenizer<> tokenizer (_);
+            int numToks = 0;
+            uint256 id;
+            std::string friendlyName;
+            for(auto const& curTok : tokenizer)
+            {
+                ++numToks;
+                if (numToks > numExpectedToks)
+                {
+                    continue;  // continue counting so can report number in the
+                               // error
+                }
+
+                if (numToks == 1)
+                    id.SetHex (curTok);
+                else
+                    friendlyName = curTok;
+            }
+
```

Comment:
Interesting.  I'm about to pick a nit, since it looks like this code should work just fine.  The inner (`curTok`) `for` loop can handle an arbitrary number of arguments, but always expects 2.  Because this code is organized as a `for` loop you have two variables (id and friendlyName) that must be default constructed and then initialized separately.  The advantage of the `for` loop is that you can count the total number of parameters passed in case there is an error.

You might consider an implementation that runs in a straight line, without the loop.  Read the `id`.  Then read the `friendlyName`.  If there are leftovers throw an error.  I'm over-simplifying a bit, since you do have to check the tokenizer iterator at each step.  The simplification may be a wash.  And your `for` loop has the advantage that it is easier to extend than a straight-line implementation.